File: 模拟/截取/外观数列/外观数列.py
# https://leetcode.cn/problems/count-and-say/

class Solution:
    def getNext(self, s):
        l, c, r = '', 1, ''
        for i in s:
            if i != l:
                if l != '':
                    r += str(c) + l
                l = i
                c = 1
            else:
                c += 1
        r += str(c) + l
        return r

    def countAndSay(self, n: int) -> str:
        s = '1'
        while n > 1:
            s = self.getNext(s)
            n -= 1
        return s

File: 模拟/截取/外观数列/外观数列.cpp
// https://leetcode.cn/problems/count-and-say/

class Solution {
public:
    string getNext(string s)
    {
        string res;
        int cot = 1, last = 0;
        for (char c : s)
        {
            if (c != last)
            {
                if (last)
                {
                    res += to_string(cot);
                    res.push_back(last);
                }
                cot = 1;
            }
            else cot ++;
            last = c;
        }
        res += to_string(cot);
        res.push_back(last);
        return res;
    }
    string countAndSay(int n) {
        string s = "1";
        while (n --> 1) s = getNext(s);
        return s;
    }
};